virttest/qemu_devices/qcontainer.py

```diff
@@ -2079,6 +2082,20 @@ def images_define_by_params(self, name, image_params, media=None,
         :param name: Name of the new disk
         :param params: Disk params (params.object_params(name))
         """
+        def _create_image_offline_snapshot():
+            """Create a offline snapshot for image file."""
+            base_img_filename = storage.get_image_filename(
+                    image_params, data_root)
+            img_format = image_params.get('image_format')
+            sn_name = 'vl.%s' % utils_misc.generate_random_string(6)
+            img_sn_filename = os.path.join('/var/tmp', sn_name)
+            qemu_img_bin = utils_misc.get_qemu_img_binary(image_params)
+            cmd = ' '.join((qemu_img_bin, 'create', '-f', img_format,
+                            img_sn_filename, '-b', base_img_filename))
```

Comment:
Have to check if the path is used by another image before create, if yes it should pick another one.

Reply:
Thanks, have updated.